### PR TITLE
fix: Fix arrow error when sorting on empty batch

### LINKED ISSUE #270: Got arrow error when sorting on empty batch
### Describe the bug

Found the issue while working on #250: https://github.com/apache/arrow-datafusion-comet/actions/runs/8681807652/job/23805034669?pr=250

For example, `SQLWindowFunctionSuite` has test failure like:

```
[info] - SPARK-7595: Window will cause resolve failed with self join *** FAILED *** (110 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 161.0 failed 1 times, most recent failure: Lost task 0.0 in stage 161.0 (TID 110) (7d92ed5170bf executor driver): org.apache.comet.CometNativeException: Arrow error: Invalid argument error: must either specify a row count or at least one column
```

There are two places that we need to fix. One is in Comet, one is in DataFusion.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #270.

### CODE REVIEW

File: core/Cargo.toml
Comment (viirya) [inline core/Cargo.toml:69]: This points to the upstream fix: https://github.com/apache/arrow-datafusion/pull/10094

**Comment by viirya** [comment]
cc @sunchao @andygrove

File: core/Cargo.toml
Comment (sunchao) [inline core/Cargo.toml:69]: hmm, should we wait until the DF PR is merged and use `https://github.com/apache/arrow-datafusion.git` instead of your personal fork?

Comment (viirya) [reply core/Cargo.toml:69]: It is pointed to my fork of DataFusion not just for this. We uses the fork of arrow-rs to use a workaround for the Java Arrow bug. So we must use the fork too in our DataFusion crate, otherwise rust compiler will complain duplicated structs etc.

Comment (sunchao) [reply core/Cargo.toml:69]: I see. I missed https://github.com/apache/arrow-datafusion-comet/pull/239. It is OK as long as we don't stay in this state for too long 😂

**Review by sunchao — APPROVED** [review]

File: core/Cargo.toml
Comment (viirya) [reply core/Cargo.toml:69]: Yea, we will change back to official arrow-rs and DataFusion once we get new Java Arrow release.

**Comment by viirya** [comment]
I will hold this until the DataFusion patch is merged. Thanks.

**Review by andygrove — APPROVED** [review]

File: core/src/execution/operators/copy.rs
Comment (viirya) [reply core/src/execution/operators/copy.rs:154]: Good catch. I will add it in this PR as it is a simple change.

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**
